xdem/coreg.py

```diff
@@ -1211,296 +1234,39 @@ def _fit_func(
-
-        assert residual < 1000, f"ICP coregistration failed: residual={residual}, threshold: 1000"
-
-        self._meta["centroid"] = centroid
-        self._meta["matrix"] = matrix
-
-
-class Deramp(Coreg):
-    """
-    Polynomial DEM deramping.
-
-    Estimates an n-D polynomial between the difference of two DEMs.
-    """
-
-    def __init__(self, degree: int = 1, subsample: int | float = 5e5) -> None:
-        """
-        Instantiate a deramping correction object.
-
-        :param degree: The polynomial degree to estimate. degree=0 is a simple bias correction.
-        :param subsample: Factor for subsampling the input raster for speed-up.
-            If <= 1, will be considered a fraction of valid pixels to extract.
-            If > 1 will be considered the number of pixels to extract.
-
-        """
-        self.degree = degree
-        self.subsample = subsample
-
-        super().__init__()
-
-    def _fit_func(
-        self,
-        ref_dem: NDArrayf,
-        tba_dem: NDArrayf,
-        transform: rio.transform.Affine,
-        crs: rio.crs.CRS,
-        weights: NDArrayf | None,
-        verbose: bool = False,
-    ) -> None:
-        """Fit the dDEM between the DEMs to a least squares polynomial equation."""
-        ddem = ref_dem - tba_dem
-        x_coords, y_coords = _get_x_and_y_coords(ref_dem.shape, transform)
-        fit_ramp, coefs = deramping(
-            ddem, x_coords, y_coords, degree=self.degree, subsample=self.subsample, verbose=verbose
-        )
-
-        self._meta["coefficients"] = coefs[0]
-        self._meta["func"] = fit_ramp
-
-    def _apply_func(
-        self, dem: NDArrayf, transform: rio.transform.Affine, crs: rio.crs.CRS, **kwargs: Any
-    ) -> tuple[NDArrayf, rio.transform.Affine]:
-        """Apply the deramp function to a DEM."""
-        x_coords, y_coords = _get_x_and_y_coords(dem.shape, transform)
-
-        ramp = self._meta["func"](x_coords, y_coords)
-
-        return dem + ramp, transform
-
-    def _apply_pts_func(self, coords: NDArrayf) -> NDArrayf:
-        """Apply the deramp function to a set of points."""
-        new_coords = coords.copy()
-
-        new_coords[:, 2] += self._meta["func"](new_coords[:, 0], new_coords[:, 1])
-
-        return new_coords
-
-    def _to_matrix_func(self) -> NDArrayf:
-        """Return a transform matrix if possible."""
-        if self.degree > 1:
-            raise ValueError(
-                "Nonlinear deramping degrees cannot be represented as transformation matrices."
-                f" (max 1, given: {self.degree})"
-            )
-        if self.degree == 1:
-            raise NotImplementedError("Vertical shift, rotation and horizontal scaling has to be implemented.")
-
-        # If degree==0, it's just a bias correction
-        empty_matrix = np.diag(np.ones(4, dtype=float))
-
-        empty_matrix[2, 3] += self._meta["coefficients"][0]
-
-        return empty_matrix
-
-
 class CoregPipeline(Coreg):
```

Comment:
This should be renamed to just `Pipeline` or `CorrPipeline` to reflect that it may not only be for co-registration, right? It may also be for bias correction (and filters in the future)

Reply:
Agreed, will bring it up in summary reply below for us to choose names! :slightly_smiling_face: